### PR TITLE
Add doc

### PR DESCRIPTION
**Pull request by JulReinhardt**
Add doc string explanation to all arguments. Some input needed, see TODO

### CODE REVIEW

File: creator/nx_creator_xpcs.py [hunk 1]
```diff
@@ -176,7 +175,25 @@ def create_xpcs_group(self,
                           **kwargs):
         """
         See NXxpcs definition for details
-        TODO: add documentation
+
+        # TODO fix/extend documentation
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:179]: Maybe could remove this TODO line and wait for issues to fix specific lines?

File: creator/nx_creator_xpcs.py [hunk 2]
```diff
@@ -176,7 +175,25 @@ def create_xpcs_group(self,
...
+        :param : g2 are values for the one-dimensional correlation as function of delay time (tau)
+        :param : g2_unit unit for g2 is usually arbitrary units
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:181]: ` unit for g2 is usually "a.u." (arbitrary units)`

File: creator/nx_creator_xpcs.py [hunk 3]
```diff
@@ -176,7 +175,25 @@ def create_xpcs_group(self,
...
+        :param : g2_stderr are the standard deviation error values for the g2 correlation values
+        :param : g2_partials_twotime is the sum across the diagonals of C2
+        :param : g2_partials_twotime_unit are subsets of the sum across the diagonals of C2
+        :param : C2 is the two-dimensional Twotime correlation
+        :param : C2_unit unit for C2 is usually arbitrary units
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:186]: as g2_unit above

File: creator/nx_creator_xpcs.py [hunk 4]
```diff
@@ -176,7 +175,25 @@ def create_xpcs_group(self,
...
+        :param : tau is the delay time corresponding to the g2 correlation values
+        :param : tau_unit any unit of time
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:188]: `any time units`

**Review by prjemian — CHANGES_REQUESTED** [review]
Some wording to be changed as noted.  Also, consistency in use of `unit` *v.* `units`. In many cases, the text might be a single value, such `nm` or `erg` (clearly singular).  In other cases, the units could involve some mathematic representation, such as `dyne*cm` or `s^-1 * m^-2 sr^-1` (clearly plural).

NeXus chose the plural use `units` which is consistent with conversational use they observed.

**Comment by JulReinhardt** [comment]
Thanks @prjemian please have a look a the changes.

File: loader/nx_loader_aps.py [hunk 5]
```diff
@@ -48,18 +47,18 @@ def xpcs_md(self):
...
-            I_partial_unit='a.u.'
+            I_partial_units='a.u.'
         )
         return saxs1d
 
     def saxs2d_md(self):
         saxs2d = dict(
             I=self.data_file.get("/exchange/pixelSum"),
-            i_unit='a.u.'
+            i_units='a.u.'
```
Comment (prjemian) [inline loader/nx_loader_aps.py:61]: Why not `I_units`?  (upper case to match `I` field)

File: creator/nx_creator_xpcs.py [hunk 6]
```diff
@@ -104,35 +104,46 @@ def _check_unit(self, name, expected, supplied):
                 user.to(expected)
                 return True
             except pint.DimensionalityError:
-                logger.warning("WARNING: '%s': Supplied unit (%s) does not match expected units (%s)",
+                logger.warning("WARNING: '%s': Supplied units (%s) does not match expected units (%s)",
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:107]: `does` -> `do`

File: creator/nx_creator_xpcs.py [hunk 7]
```diff
@@ -76,24 +76,24 @@ def _create_dataset(self, group, name, value, **kwargs):
...
+        :param expected: expected units
+        :param supplied: units string that was supplied
         :return *bool*: `True` if units conversion is possible:
         """
 
-        # catch arbitrary unit separately from pint --> point that out in documentation
+        # catch arbitrary units separately from pint --> point that out in documentation
         if supplied in ['au', 'a.u.', 'a.u']:
-            logger.info("Info: arbitrary units supplied for '%s' in form of '%s' np unit conversion applicable",
+            logger.info("Info: arbitrary units supplied for '%s' in form of '%s' np units conversion applicable",
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:96]: `np` -> `no`

**Review by prjemian — CHANGES_REQUESTED** [review]
changes as noted

**Comment by ambarb** [comment]
@JulReinhardt is this still needed for NXxpcs V1?

**Comment by prjemian** [comment]
@JulReinhardt : Looks like it's time for me to review again, right?

**Comment by ambarb** [comment]
This is for both the creator and NXdefinition.  Ready for review.

We merge after review if no glaring issues.  Wait for XPCS community comments before addressing.

**Review by prjemian — APPROVED** [review]

**Pull request merged**